packages/template-retail-react-app/app/pages/checkout/util/checkout-context.js

```diff
@@ -70,13 +70,13 @@ export const CheckoutProvider = ({children}) => {
             mergeState({isGuestCheckout: false})
         }
 
-        if (customer.isGuest && basket.customerInfo?.email && !state.isGuestCheckout) {
+        if (customer.isGuest && basket?.customerInfo?.email && !state.isGuestCheckout) {
             mergeState({isGuestCheckout: true})
         }
 
         // Derive the starting step for checkout based on current state of basket.
         // A failed condition sets the current step and returns early (order matters).
-        if (customer.customerId && basket.basketId && state.step == undefined) {
+        if (customer.customerId && basket?.basketId && state.step == undefined) {
```

Comment:
There is no reason the step can be null 
```suggestion
        if (customer.customerId && basket?.basketId && state.step === undefined) {
```